File: Ser2023/Company/Company.Web/Modules/ServerTypes/Administration/TenantRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface TenantRow {
    TenantId?: number;
    TenantName?: string;
}

export abstract class TenantRow {
    static readonly idProperty = 'TenantId';
    static readonly nameProperty = 'TenantName';
    static readonly localTextPrefix = 'Administration.Tenant';
    static readonly lookupKey = 'Administration.Tenant';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TenantRow>('Administration.Tenant') }
    static async getLookupAsync() { return getLookupAsync<TenantRow>('Administration.Tenant') }

    static readonly deletePermission = 'Administration:Tenants';
    static readonly insertPermission = 'Administration:Tenants';
    static readonly readPermission = 'Administration:Tenants';
    static readonly updatePermission = 'Administration:Tenants';

    static readonly Fields = fieldsProxy<TenantRow>();
}

File: Ser2023/Company/Company.Web/Modules/Administration/Tenant/TenantRow.cs

using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace Company.Administration
{
    [ConnectionKey("Default"), Module("Administration"), TableName("[dbo].[Tenants]")]
    [DisplayName("Tenants"), InstanceName("Tenant"), TwoLevelCached]
    [ReadPermission(PermissionKeys.Tenants)]
    [ModifyPermission(PermissionKeys.Tenants)]
    [LookupScript("Administration.Tenant")]
    public sealed class TenantRow : Row<TenantRow.RowFields>, IIdRow, INameRow
    {
        [DisplayName("Tenant Id"), Identity, IdProperty]
        public Int32? TenantId
        {
            get => Fields.TenantId[this]; 
            set => Fields.TenantId[this] = value; 
        }

        [DisplayName("Tenant Name"), Size(100), NotNull, QuickSearch, NameProperty]
        public String TenantName
        {
            get => Fields.TenantName[this]; 
            set => Fields.TenantName[this] = value; 
        }

        public TenantRow()
        {
        }

        public TenantRow(RowFields fields)
            : base(Fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field TenantId;
            public StringField TenantName;
        }
    }
}
